src/helix-ui/elements/HXFileTileElement.js

```diff
@@ -0,0 +1,127 @@
+import { HXElement } from './HXElement';
+import shadowMarkup from './HXFileTileElement.html';
+import shadowStyles from './HXFileTileElement.less';
+
+export class HXFileTileElement extends HXElement {
+    static get is () {
+        return 'hx-file-tile';
+    }
+
+    static get template () {
+        return `<style>${shadowStyles}</style>${shadowMarkup}`;
+    }
+
+    constructor () {
+        super();
+        this._onDelete = this._onDelete.bind(this);
+    }
+
+    $onConnect () {
+        this.$upgradeProperty('icon');
+        this.$upgradeProperty('name');
+        this.$upgradeProperty('href');
+        this._btnDelete.addEventListener('click', this._onDelete);
+    }
+
+    $onDisconnect () {
+        this._btnDelete.removeEventListener('click', this._onDelete);
+    }
+
+    static get $observedAttributes () {
+        return [ 
+            'icon', 
+            'name',
+            'href',
+        ];
+    }//$observedAttributes
+
+    $onAttributeChange (attr, oldVal, newVal) {
+        switch (attr) {
+            case 'icon':
+                this._elIcon.type = newVal;
+                break;
+
+            case 'href':
+                if (newVal !== null) {
+                    this._elLink.href = newVal;
+                } else {
+                    this._elLink.removeAttribute('href');
+                }
+                break;    
+
+            case 'name':
+                this._elLink.innerText = newVal;
+                this._elExt.innerText = this._extension || '';
+                break;   
+        }
+    }//$onAttributeChange
+
+    /**
+     * Programmatically delete the file tile (removes element from the DOM).
+     */
+    delete () {
+        this.remove();
+    }
+
+    // PRIVATE METHODS
+    _onDelete (evt) {
+        evt.preventDefault();
+
+        if (this.$emit('delete')) {
```

Comment:
add logic to emit _either_ `delete` (when in default state) or `cancel` when loading or invalid